frost-core/src/frost/keys.rs

```diff
@@ -329,12 +242,28 @@ pub fn keygen_with_dealer<C: Ciphersuite, R: RngCore + CryptoRng>(
     let mut bytes = [0; 64];
     rng.fill_bytes(&mut bytes);
 
-    let secret = SharedSecret::random(rng);
-    let group_public = VerifyingKey::from(&secret);
+    let key = SigningKey::new(rng);
+
+    split_key(&key, max_signers, min_signers, rng)
+}
+
+/// Splits an existing key into FROST shares.
+///
+/// This is identical to [`keygen_with_dealer`] but receives an existing key
+/// instead of generating a fresh one. This is useful in scenarios where
+/// the key needs to be generated externally or must be derived from e.g. a
+/// seed phrase.
+pub fn split_key<C: Ciphersuite, R: RngCore + CryptoRng>(
```

Comment:
What about just `split()`? (Non-blocking, just thinking)

Reply:
Sounds good, done in 5346ec29